ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+
+
+import glob
+import re
+
+import uproot
+import numpy as np
+import scipy.interpolate
+
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.io.eventsource import EventSource
+from ctapipe.io.containers import DataContainer, TelescopePointingContainer, WeatherContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription, OpticsDescription, CameraGeometry
+import gzip
+import struct
+
+__all__ = ['MAGICEventSourceHDF5', 'MAGICEventSourceROOT']
+
+
+class MAGICEventSourceHDF5(EventSource):
+    """
+    EventSource for MAGIC raw data converted to hdf5.
+
+    This class utilises `h5py` to read the hdf5 file, and stores the
+    information into the event containers.
+    """
+    _count = 0
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            import h5py
+        except ImportError:
+            msg = "The `h5py` python module is required to access the MAGIC data or MCs"
+            self.log.error(msg)
+            raise
```

Comment:
`h5py` is a default anaconda package In Py3.5 and Py3.6 anaconda for Linux (both 32bit and 64bit versions). 
One can find a list of all packages coming with anaconda here: 

https://docs.anaconda.com/anaconda/packages/py3.6_linux-64/

In addition ctapipe has made `h5py` an explicit dependency by stating it in the `environment.yml`:

https://github.com/cta-observatory/ctapipe/blob/master/environment.yml#L14

So I think there is no need to treat h5py as an optional dependency here.